File: php/src/Model/WorkflowRunPage.php
<?php

declare(strict_types=1);

namespace Fivexer\SDK\Model;

use Fivexer\SDK\Internal\Json;

/**
 * A cursor-paginated page of runs.
 *
 * Response model: built from a decoded JSON array and read-only thereafter.
 */
final class WorkflowRunPage
{
    public function __construct(
        /** @var list<WorkflowRun> */
        public readonly array $runs,
        public readonly ?string $nextCursor,
    ) {
    }

    /**
     * @param array<string, mixed> $data
     * @internal
     */
    public static function fromArray(array $data): self
    {
        return new self(
            runs: Json::parseEach($data, 'runs', [WorkflowRun::class, 'fromArray']),
            nextCursor: isset($data['nextCursor']) ? (string) $data['nextCursor'] : null,
        );
    }
}

File: java/src/main/java/io/fivexer/sdk/model/WorkflowRunPage.java
package io.fivexer.sdk.model;

import java.util.List;

/**
 * A cursor-paginated page of runs.
 *
 * <p>Populated by Gson via field reflection and exposed read-only; callers receive instances
 * from the client and never construct them.
 */
public class WorkflowRunPage {
    private List<WorkflowRun> runs;
    private String nextCursor;

    public List<WorkflowRun> getRuns() { return runs; }
    public String getNextCursor() { return nextCursor; }
}
